system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp

```diff
@@ -82,28 +82,26 @@ void MrmHandler::onOperationModeAvailability(
   const tier4_system_msgs::msg::OperationModeAvailability::ConstSharedPtr msg)
 {
   stamp_operation_mode_availability_ = this->now();
+  operation_mode_availability_ = msg;
+  const bool skip_emergency_holding_check =
+    !param_.use_emergency_holding || is_emergency_holding_ || !isOperationModeAutonomous();
 
-  if (!param_.use_emergency_holding) {
-    operation_mode_availability_ = msg;
+  if (skip_emergency_holding_check) {
     return;
   }
 
-  if (!is_emergency_holding_) {
-    if (msg->autonomous) {
-      stamp_autonomous_become_unavailable_.reset();
-    } else {
-      if (!stamp_autonomous_become_unavailable_.has_value()) {
-        stamp_autonomous_become_unavailable_.emplace(this->now());
-      } else {
-        const auto emergency_duration =
-          (this->now() - stamp_autonomous_become_unavailable_.value()).seconds();
-        if (emergency_duration > param_.timeout_emergency_recovery) {
-          is_emergency_holding_ = true;
-        }
-      }
-    }
+  if (msg->autonomous) {
+    stamp_autonomous_become_unavailable_.reset();
+    return;
   }
-  operation_mode_availability_ = msg;
+
+  stamp_autonomous_become_unavailable_ = (!stamp_autonomous_become_unavailable_.has_value())
```

Comment:
I prefer the following. What do you think?

```
if (stamp_autonomous_become_unavailable_.has_value()) {
  const auto emergency_duration =
    (this->now() - stamp_autonomous_become_unavailable_.value()).seconds();
  is_emergency_holding_ = (emergency_duration > param_.timeout_emergency_recovery);
} else {
  stamp_autonomous_become_unavailable_.emplace(this->now());
}

Reply:
@veqcc I don't think it is worth it to add branching to the code, it decreases readability and increases cyclomatic complexity. Adding the else branch only skips these checks: 

`  const auto emergency_duration =
    (this->now() - stamp_autonomous_become_unavailable_.value()).seconds();
  is_emergency_holding_ = (emergency_duration > param_.timeout_emergency_recovery);`

which are not costly to compute. Also the cost of computing these operations might be lower than having a branching code. I believe Adding comments is ok, but dont agree with using an else branch if it is not necessary.

Reply:
Agreed :+1: